### PR TITLE
Add packit support

### PR DESCRIPTION
**Pull request by ianballou**
My biggest question mark currently is what needs to happen for `fix-spec-file` if anything.

### CODE REVIEW

File: .packit.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  post-upstream-clone:
+    - "wget https://raw.githubusercontent.com/theforeman/foreman-packaging/rpm/develop/packages/plugins/rubygem-smart_proxy_container_gateway/rubygem-smart_proxy_container_gateway.spec -O rubygem-smart_proxy_container_gateway.spec"
+  get-current-version:
+    - ruby -rrubygems -e 'puts Gem::Specification::load(Dir.glob("*.gemspec").first).version'
+  create-archive:
+    - gem build smart_proxy_container_gateway.gemspec
+    - bash -c "ls -1t ./smart_proxy_container_gateway-*.gem | head -n 1"
+    #fix-spec-file:
+    #- 'bash -c "sed -i \"s/Source0:.*/Source0: ${PACKIT_PROJECT_ARCHIVE}/\" rubygem-smart_proxy_container_gateway.spec"'
+    #- bash -c 'sed -i "/global release/a%global nightly ${PACKIT_RPMSPEC_RELEASE:1}" rubygem-smart_proxy_container_gateway.spec'
```
Comment (ekohl) [inline .packit.yaml:28]: I think this is only needed for nightly packages and can be dropped safely.

File: .packit.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# See the documentation for more information:
+# https://packit.dev/docs/configuration/
+
+specfile_path: rubygem-smart_proxy_container_gateway.spec
+
+# add or remove files that should be synced
+files_to_sync:
+    - rubygem-smart_proxy_container_gateway.spec
+    - .packit.yaml
```
Comment (ekohl) [inline .packit.yaml:9]: Can we consistently indent this?

File: .packit.yaml (hunk 1 above)
Comment (ianballou) [reply .packit.yaml:28]: yeah, seems not needed :+1:

File: .packit.yaml (hunk 2 above)
Comment (ianballou) [reply .packit.yaml:9]: Ha probably because it's copy and pasted all over the organization. I'll fix em.

**Comment by ianballou** [comment]
Seems to work just fine, so I'm merging to enable builds on current PRs.

**Pull request merged**